### PR TITLE
    BF: close out #1484.  Return type of unordered_map erase methods …

### PR DESCRIPTION
**Pull request by MichaelWS**
…should be iterator not void

### CODE REVIEW

**Comment by scoder** [comment]
You can add the `-std` cflag for all `cpp11`/`cpp17`/... tagged tests through this mechanism:
https://github.com/cython/cython/blob/0.27.1/runtests.py#L313
And by returning `EXCLUDE_EXT` from the update function, you can make the test module be ignored. See
https://github.com/cython/cython/blob/0.27.1/runtests.py#L232
https://github.com/cython/cython/blob/0.27.1/runtests.py#L922

File: runtests.py [hunk 1]
```diff
@@ -249,6 +248,38 @@ def update_openmp_extension(ext):
...
+    return EXCLUDE_EXT
+
+
+def get_gcc_version():
+    """
+        finds gcc version using Popen
+    """
+    matcher = re.compile(r"gcc version (\d+\.\d+)").search
+    try:
+        p = subprocess.Popen([cc, "-v"], stderr=subprocess.PIPE, env=env)
```
Comment (scoder) [inline runtests.py]: `cc` is undefined. You need to also move over the initial setup from the original function (CC/env/etc.).

**Comment by MichaelWS** [comment]
Thank you

File: runtests.py [hunk 2]
```diff
@@ -261,10 +261,26 @@ def update_cpp11_extension(ext):
...
         finds gcc version using Popen
     """
+    if language == 'cpp':
+        cc = sysconfig.get_config_var('CXX')
+    else:
+        cc = sysconfig.get_config_var('CC')
+
+    if not cc:
+        if sys.platform == 'win32':
+            return '/openmp', ''
```
Comment (scoder) [inline runtests.py]: This was a bit too much.

**Comment by MichaelWS** [comment]
Yes indeed, was moving too fast.    I will clean up the git history and
squash once the test is running correctly.

Also, do the tests have to return something?

**Comment by scoder** [comment]
> do the tests have to return something?

No, assertions are just fine. As I said, it's a doctest, so anything that you return from a call will also be checked, in addition to anything you do in the function you call.

I think on travis `CC` and `CXX` are not set, so it currently hits the `return None` line, which makes the subsequent `float()` call fail with a `TypeError`. I think we'll need some kind of callback for that case. The best way to figure out the compiler is probably still to leave it to distutils. It has all the machinery in place. Though it's pretty good in hiding it ;)

**Comment by MichaelWS** [comment]
great. modified to use distutils

---
Michael


-----Original Message-----

Re: [cython/cython] BF: close out #1484. Return type of unordered_map erase methods &hellip; (#1918)

**Comment by MichaelWS** [comment]
I am cleaning up all the commits into one

**Pull request closed without merging**